File: hw1/datatxt/insertion.py
def insertionSort(array):
	for i in range(1, len(array)):
		val = array[i]
		position = i - 1

		while (position >= 0) and (array[position] > val):
			array[position + 1] = array[position]
			position = position - 1

		array[position + 1] = val


arr = []

with open('data.txt') as f:
	for line in f:
		arr.append([int(v) for v in line.split()])

fo = open("insertion.out", "wb")
for sub in range(0, len(arr)):
	insertionSort(arr[sub])
	fo.writelines(str(arr[sub]))
	fo.write("\n")

fo.close()

File: hw1/insertion.cpp
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>
using namespace std;

const int MAX = 9999;

void insertion(int array[], int n){
	int i, j, point;
	for(i = 1; i < n; i++){
		point = array[i];
		j = i - 1;
		while(j >= 0 && array[j] > point){
			array[j + 1] = array[j];
			j = j - 1;
		}
		array[j + 1] = point;
	}
}

int main(){
	ifstream inFile;

	int n = 0;
	int arr[MAX];
	string line;
	int ticker = 0;

	inFile.open("data.txt");

	//while( !inFile.eof()){
	while(inFile >> arr[n]){
		getline(inFile, line);
		//ticker = stoi(line);
		stringstream stream(line);
		
		while(1){
			int j;
			stream >> j;
			if(!stream){
				break;
			}
			arr[ticker] = j;
			ticker++;
		}
		//insertion(arr, ticker);
		for(int i = 0; arr[i]; i++){
			cout << "Val ( " << n << ") " << arr[i];
		}
		//SEND TO OUTFILE HERE
		n++;
	}

	//inFile >> arr[n];
	//while(inFile){
	//	n++;
	//	inFile >> arr[n];
	//}
	
	inFile.close();

	//for(int i = 0; arr[i]; i++){
	//	cout << arr[i];
	//}
	

	//insertion(arr, n);

	//inFile.close();

	return 0;
}
